src/containers/SignIn/index.js

```diff
@@ -25,19 +25,20 @@ class SignInForm extends React.Component {
       form: { validateFields },
     } = this.props;
     e.preventDefault();
-    validateFields((err, values) => {
+    validateFields(async (err, values) => {
       if (!err) {
         this.setState({ loading: true });
-        firebase
-          .doSignInWithEmailAndPassword(values.email, values.password)
-          .then(res => {
-            localStorage.setItem('userId', res.user.uid);
-            this.setState({ loading: false });
-            history.push(ROUTES.HOME);
-          })
-          .catch(error => {
-            this.setState({ error, loading: false });
-          });
+        try {
+          const result = await firebase.doSignInWithEmailAndPassword(
+            values.email,
+            values.password
+          );
+          await localStorage.setItem('userId', result.user.uid);
```

Comment:
We don't really need to use await with local storage here, await is used **to wait for promises to be resolved** and local storage is not a promise.
- This line of code will only happen after the upper await finishes.